webapp/redux/middleware/plotMiddleware.js

```diff
@@ -1,38 +1,36 @@
-import { ADD_WIDGET, addWidget, SET_WIDGETS, UPDATE_WIDGET } from '../actions/layout';
+import { addWidget, SET_WIDGETS, UPDATE_WIDGET } from '../actions/layout';
 import Utils from '../../Utils';
-import { NETPYNE_COMMANDS, PLOT_WIDGETS, MODEL_STATE } from '../../constants';
-
+import { NETPYNE_COMMANDS, PLOT_WIDGETS, WidgetStatus } from 'root/constants';
 import { processError } from './middleware';
-import { SIMULATE_NETWORK, CREATE_SIMULATE_NETWORK, SET_THEME } from '../actions/general';
-
-// Cache for plots coming from the backend
-window.plotSvgImages = {};
-
+import { SIMULATE_NETWORK, CREATE_SIMULATE_NETWORK, SET_THEME, CREATE_NETWORK } from '../actions/general';
 
 export default store => next => action => {
 
   async function setWidget (widget) {
-  
+
     const { plotMethod, plotType } = widget.method;
-    return plotFigure(widget.id, plotMethod, plotType, store.getState().general.theme).then(result => {
-      if (!result) {
-        console.warn('Plot not retrieved:', widget.id);
-        widget.disabled = true;
-        return widget;
-        // return null;
-      } else {
-        console.debug('Plot retrieved:', widget.id);
-        widget.disabled = false;
-        return widget;
-      }
-    });
-  
+    return plotFigure(widget.id, plotMethod, plotType, store.getState().general.theme)
+      .then(result => {
+        widget.initialized = true;
+        if (result) {
+          console.debug('Plot retrieved:', widget.id);
+          widget.disabled = false;
+          widget.data = result;
```

Comment:
I dont' remember what the data really is here. If it's the html, the redux store is not the right place to put it (maybe that's why we had the cache variable)